File: tests/integration/benchmarks/fixtures/java/rosetta/src/com/rosetta/models/Record.java
package com.rosetta.models;

import com.rosetta.types.Entry;
import com.rosetta.types.Status;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

/**
 * A data record to be processed.
 */
public class Record extends Entry {
    private final double value;
    private final List<String> tags;

    public Record(String id, double value) {
        super(id, Instant.now(), Status.PENDING);
        this.value = value;
        this.tags = new ArrayList<>();
    }

    /**
     * Static factory for creating records (used by tests).
     */
    public static Record create(int index, String id, double value) {
        return new Record(id, value);
    }

    public double getValue() {
        return value;
    }

    public List<String> getTags() {
        return tags;
    }
}

File: tests/integration/benchmarks/fixtures/csharp/rosetta/src/Models/Record.cs
using Rosetta.Types;

namespace Rosetta.Models;

/// <summary>
/// A data record to be processed.
/// </summary>
public class Record : Entry
{
    public double Value { get; }
    public List<string> Tags { get; }

    public Record(string id, double value) 
        : base(id, DateTime.UtcNow, Status.Pending)
    {
        Value = value;
        Tags = new List<string>();
    }

    /// <summary>
    /// Factory method for creating records with sensible defaults.
    /// </summary>
    public static Record Create(int id, string name, double value) => 
        new($"{name}-{id}", value);

    /// <summary>
    /// Validates configuration is within acceptable bounds.
    /// Delegates to ModelFactory.ValidateConfig for actual validation.
    /// </summary>
    public static bool ValidateConfig(ProcessorConfig config) =>
        ModelFactory.ValidateConfig(config);
}
